miner/scroll_worker.go

```diff
@@ -488,7 +488,7 @@ func (w *worker) newWork(now time.Time, parent *types.Block, reorging bool, reor
 	// Set baseFee if we are on an EIP-1559 chain
 	if w.chainConfig.IsCurie(header.Number) {
 		parentL1BaseFee := fees.GetL1BaseFee(parentState)
-		header.BaseFee = misc.CalcBaseFee(w.chainConfig, parent.Header(), parentL1BaseFee)
+		header.BaseFee = misc.CalcBaseFee(w.chainConfig, parent.Header(), parentL1BaseFee, header.Time)
```

Comment:
There's a bit of a concern here around the fork transition:

`header.Time` is set to `now` here, but it might be overwritten by `w.engine.Prepare`. I think it can only decrease if we have relaxed block times enabled. So maybe not a concern in practice.

Reply:
I added a lot of special logic to handle the EuclidV2 transition in `func (ue *UpgradableEngine) Prepare`. But you're right, I think timestamp won't decrease after `Prepare`.

Reply:
Yeah. So think that it's kind of complex and also unnecessary to consider `euclidv2` + `feynman` case, assuming `feynman` would happen at least a moment after `euclidv2` transition is completed.